src/user/mod.rs

```diff
@@ -186,12 +184,14 @@ impl<'a> From<&'a UserID<'a>> for UserID<'a> {
 /// * `show_all_inline_media`
 /// * `time_zone`/`utc_offset`
 /// * `withheld_in_countries`/`withheld_scope`
-#[derive(Debug, Clone)]
+// TODO codepoints_to_bytes on entity ranges
```

Comment:
Okay, now i see how awkward this gets. Still, would it be possible to have some "raw" versions of the TwitterUser struct, like you did for Tweet? Maybe it could just have a conversion function from the "raw" version to the final one that calls `codepoints_to_bytes` there, since it seems like that's the only processing the original `FromJson` implementation did on top of extracting the fields. Then the `TwitterUser` impl would just defer to `RawUser` and call the conversion function.

I want to avoid having codepoint-based indices in the public API, partly because some of the byte-based indices (in `Tweet`) were able to stay, but also because indexing by codepoints is so much more tedious in Rust. The problem with having some function on `UrlEntity` (or the like) that needs to be called to get the proper string slice (or indices) is that the entity struct (by design and necessity) doesn't have the source string, so it would need to be passed in. It would also be awkward for it to be on the user, because there are loads of entities sitting on the user, so the indices would need to be passed in there. This is just a really awkward situation.